native_client/ctcdecode/scorer.cpp

```diff
@@ -134,8 +136,10 @@ void Scorer::save_dictionary(const std::string& path)
   fout.write(reinterpret_cast<const char*>(&MAGIC), sizeof(MAGIC));
   fout.write(reinterpret_cast<const char*>(&FILE_VERSION), sizeof(FILE_VERSION));
   fout.write(reinterpret_cast<const char*>(&is_character_based_), sizeof(is_character_based_));
-  fst::FstWriteOptions opt;
-  dictionary->Write(fout, opt);
+  if (!is_character_based_) {
```

Comment:
So technically we will still produce a trie file :)

Reply:
Yep. My reasoning is that it simplifies the documentation, everyone follows the same steps, even if for character based LMs the trie file will be a tiny dummy file. Also simplifies the code changes.

Reply:
100% agree with you